File: app/src/main/java/com/uvnode/typhoon/extensions/source/HasMirrors.java
package com.uvnode.typhoon.extensions.source;

import com.uvnode.typhoon.extensions.api.ApiCallbacks;
import com.uvnode.typhoon.extensions.model.Episode;
import com.uvnode.typhoon.extensions.model.Mirror;
import com.uvnode.typhoon.extensions.model.Series;

public interface HasMirrors {

    void getVideoUris(ApiCallbacks callbacks, Series series, Episode episode, Mirror mirror);
}

File: app/src/main/java/com/uvnode/typhoon/extensions/source/HasMirrors.kt
package com.uvnode.typhoon.extensions.source

import com.uvnode.typhoon.extensions.api.ApiCallbacks
import com.uvnode.typhoon.extensions.model.Episode
import com.uvnode.typhoon.extensions.model.Mirror
import com.uvnode.typhoon.extensions.model.Series

interface HasMirrors {
    @Throws(Exception::class)
    fun getVideoUris(callbacks: ApiCallbacks, series: Series, episode: Episode, mirror: Mirror)
}